File: app/src/main/java/fudi/freddy/mokitosample/share/RepositoryShare.java
package fudi.freddy.mokitosample.share;

import fudi.freddy.mokitosample.App;

/**
 * @version 1.0
 * @autor Romell Domínguez
 * @date 8/13/17
 */

public class RepositoryShare implements Repository {

    private RepositoryShare(){

    }

    public static RepositoryShare getInstance() {
        return new RepositoryShare();
    }
    @Override
    public void save(String newData) {
        App.getPrefer().edit().putString("text",newData).commit();
    }

    @Override
    public String load() {
        return App.getPrefer().getString("text", "");
    }
}

File: app-kotlin/src/main/java/fudi/freddy/mokitosample/share/RepositoryShare.kt
package fudi.freddy.mokitosample.share

import fudi.freddy.mokitosample.App

/**
 * @version 1.0
 * @autor Romell Domínguez
 * @date 7/25/20
 */
class RepositoryShare private constructor() : Repository {

    override fun save(newData: String?) {
        App.prefer.edit().putString("text", newData).apply()
    }

    override fun load(): String? {
        return App.prefer.getString("text", "")
    }

    companion object {
        var instance: Repository  = RepositoryShare()
    }
}